OpenRA.Mods.Common/Traits/Attack/AttackTurreted.cs

```diff
@@ -35,11 +37,13 @@ protected override bool CanAttack(Actor self, Target target)
 			if (!base.CanAttack(self, target))
 				return false;
 
+			// Rotate all turrets to make sure upgrading a turret does not change the turret's facing.
+			var faceTarget = false;
 			foreach (var t in turrets)
-				if (t.FaceTarget(self, target))
-					return true;
+				if (t.FaceTarget(self, target) && !t.IsTraitDisabled)
```

Comment:
That's the point. Any other way and turretchanging would look odd (say IFV targets something, passenger exits and turret suddenly looks somewhere else).

It's also a bit workaroundish because turrets and `AttackTurret` is only paired at `Armament` level which `FaceTarget` doesn't check for, allowing the facings to go out of sync. (Although I have fixed that since, notsure.)